tests/integration/run_modes/test_mode_overrides.py

```diff
@@ -0,0 +1,169 @@
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Test that using [runtime][TASK]run mode works in each mode.
+
+Point 3 of the Skip Mode proposal
+https://github.com/cylc/cylc-admin/blob/master/docs/proposal-skip-mode.md
+
+| The run mode should be controlled by a new task configuration
+| [runtime][<namespace>]run mode with the default being live.
+| As a runtime configuration, this can be defined in the workflow
+| for development / testing purposes or set by cylc broadcast.
+
+n.b: This is pretty much a functional test and
+probably ought to be labelled as such, but uses the
+integration test framework.
+"""
+
+import pytest
+
+from cylc.flow.cycling.iso8601 import ISO8601Point
+from cylc.flow.run_modes import WORKFLOW_RUN_MODES, RunMode
+
+
+@pytest.mark.parametrize('workflow_run_mode', sorted(WORKFLOW_RUN_MODES))
+async def test_run_mode_override_from_config(
+    capture_live_submissions,
+    flow,
+    scheduler,
+    run,
+    complete,
+    workflow_run_mode,
+    validate
+):
+    """Test that `[runtime][<namespace>]run mode` overrides workflow modes."""
+    id_ = flow({
+        'scheduling': {
+            'graph': {
+                'R1': 'live & skip',
+            },
+        },
+        'runtime': {
+            'root': {'simulation': {'default run length': 'PT0S'}},
+            'live': {'run mode': 'live'},
```

Comment:
Wait so we can specify `[runtime][X]run mode = live` but it doesn't do anything? At some point wasn't this `workflow` rather than `live`?

Edit for clarity: I just tested
```cylc
[scheduling]
    [[graph]]
        R1 = foo
[runtime]
    [[foo]]
        script = sleep 120
        run mode = live
        [[[simulation]]]
            default run length = PT0S
```
```shell
cylc play workflow --mode simulation
```
The task runs and finishes instantly.

Reply:
The "run mode" setting only applies in "live" mode. Otherwise the workflow run mode takes precedence.

Reply:
It doesn't do anything if the workflow is in skip mode. If the workflow is in live mode and you've set the task to skip mode, broadcasting `run mode = live` will turn skip mode off. That it's not a just a null value is a legacy of the era when this was meant to be able to take any mode (live/dummy/sim/skip). I think it's nicer (more explicit) than broadcasting `run mode=`.

Reply:
> I don't think that it was ever "workflow" rather than "live"

You had originally implemented "workflow" as an option (in addition to "live" and all the other run modes) - https://github.com/cylc/cylc-flow/pull/6039#discussion_r1713993391

Is there a reason the config is not called `skip mode = True/False` instead of `run mode = live/skip`?

Reply:
Not any more.